layouts/_default/single.html

```
@@ -65,6 +124,176 @@ <h1 class="post-title">
+      </nav>
+      {{- else -}}
+      <p style="font-size:0.75rem;color:var(--color-ink-muted);font-style:italic;">{{ i18n "tools.toc_empty" | default "No headings" | safeHTML }}</p>
+      {{- end -}}
+    </div>
+
+    {{/* Panel 2: AI Companion */}}
+    <div class="abs-panel" id="abs-panel-ai" role="tabpanel" aria-labelledby="abs-tab-ai" hidden>
+      <div class="abs-ai-panel">
+        <div class="abs-ai-messages" aria-live="polite"></div>
+        <div class="abs-ai-input-area">
+          <textarea class="abs-ai-textarea" rows="2"
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)